src/gui/base/buttons/BannerButton.ts

```diff
@@ -0,0 +1,58 @@
+import { lang, TranslationKey } from "../../../misc/LanguageViewModel.js"
+import { lazy } from "@tutao/tutanota-utils"
+import m, { Children, Component, Vnode } from "mithril"
+import { px, size } from "../../size.js"
+import { theme } from "../../theme.js"
+import { CalendarAttendeeStatus } from "../../../api/common/TutanotaConstants.js"
+import { EventPreviewViewAttrs } from "../../../calendar/view/EventPreviewView.js"
+
+export type BannerButtonAttrs = {
+	borderColor: string
+	color: string
+	click: () => unknown
+	text: TranslationKey | lazy<string>
+}
+
+export class BannerButton implements Component<BannerButtonAttrs> {
+	view({ attrs }: Vnode<BannerButtonAttrs>): Children {
+		return m(
+			"button.border-radius.mr-s.center",
+			{
+				style: {
+					border: `2px solid ${attrs.borderColor}`,
+					background: "transparent",
+					color: attrs.color,
+					width: "min-content",
+					padding: px(size.hpad_button),
+					minWidth: "60px",
+				},
+				onclick: attrs.click,
+			},
+			lang.getMaybeLazy(attrs.text),
+		)
+	}
+}
+
+export function renderReplyButtons(participation: NonNullable<EventPreviewViewAttrs["participation"]>) {
```

Comment:
or use segmentControl maybe

Reply:
segmented control doesn't usually have "nothing selected" state, it's not it

Reply:
I guess you cold say that, but I doubt that's a general guideline. To me it would make sense to use a segment control for a choice that can be made once and only changed, but not completely taken back later

I made a pureComponent out of it and moved it to the EventPreviewView, which took some doing since mail-view and calendar-view can't import each other and putting the import all the way at the top of the mailview hierarchy just feels completely wrong.

we may need an asyncComponent utility in the future to make things like this easier, maybe with a placeholder feature so there's no UI shift even if the import takes a bit.